### DISCUSSION TITLE
cannot get react-static export working on aws lambda

### DISCUSSION THREAD

**Discussion by Davidvision** [react-static/react-static#1591] [Q&A]
Hi everybody,

I've been trying to export my react-static bundled website on aws lambda but it seems i can't get it to work, so I really hope someone will be able to help me on this :) !

Here are the steps i tried :

- I followed the steps mentioned [here](https://github.com/react-static/react-static/blob/master/docs/guides/aws-lambda.md) to trigger a build/bundle/export but none seemed to work, even while testing locally. When I run `await rs()`, I get the following error on the console :
```
Cannot read property 'config' of undefined
  
  - getConfig.js:75 getConfig
    [client]/[react-static]/lib/static/getConfig.js:75:60
  
  - bundle.js:62 _callee$
    [client]/[react-static]/lib/commands/bundle.js:62:46
  
  - runtime.js:63 tryCatch
    [client]/[regenerator-runtime]/runtime.js:63:40
  
  - runtime.js:293 Generator.invoke [as _invoke]
    [client]/[regenerator-runtime]/runtime.js:293:22
  
  - runtime.js:118 Generator.next
    [client]/[regenerator-runtime]/runtime.js:118:21

...
````

Well, after this wondered if I could get it working while running the actual react-static command as such :

```
const exec = require("child_process").exec;
exec("./node_modules/react-static/bin/react-static build");
```

Running **build**, it worked, both locally and in the lambda function where I was able to upload the resulting tmp/dist directory on my s3 bucket, awesome.

But my aim is to only do `react-static export` on lambda, as I won't bundle the app much often.

Well, I first ran `react-static bundle` locally, then I tried the following : 
```
exec("./node_modules/react-static/bin/react-static export");
```

It worked perfectly on my local machine, but it doesn't on lambda.

the error i'm getting in lambda's log is the following : 

```
WARN	Error: Command failed: ./node_modules/react-static/bin/react-static export
Creating default 404 because none was found...
Trace: {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/tmp/artifacts/static-app.js',
    '/var/task/node_modules/react-static/lib/static/exportRoutes.threaded.js'
  ]
}
    at /var/task/node_modules/react-static/bin/react-static:6:11
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:314:20)
    at ChildProcess.EventEmitter.emit (domain.js:483:12)
    at maybeClose (internal/child_process.js:1022:16)
    at Socket.<anonymous> (internal/child_process.js:444:11)
    at Socket.emit (events.js:314:20)
    at Socket.EventEmitter.emit (domain.js:483:12)
    at Pipe.<anonymous> (net.js:675:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: './node_modules/react-static/bin/react-static export'
}
```

It seems that react-static isn't able to find `'/tmp/artifacts/static-app.js'`.

I'm aware that the local task directory in lambda is a read-only one, and that the only writable one is the /tmp directory. So i tried to copy the whole artifacts, dist, and tmp directories to /tmp, making sure that config.paths in static.config.js all point to the /tmp folder, but I still get the error...

I honestly ran out of ideas on how to solve this, that's why I came here !

So...

First, any ideas on why running `await rs()` (as in [here](https://github.com/react-static/react-static/blob/master/docs/guides/aws-lambda.md)) doesn't work as it seems to be the recommended way ?

And any ideas on why react-static isn's able to find find `'/tmp/artifacts/static-app.js'` ?

Thanks a lot for your help :)

**Comment by Davidvision** [comment]
Well, I found a working solution now :)

**Reply by loginov-rocks** [reply]
@Davidvision can you share it, please? :)

**Reply by Davidvision** [reply]
@loginov-rocks I ended up using codebuild instead of lambda to export it !

**Reply by loginov-rocks** [reply]
Cheating :D I resolved it as well, in my case it was the project build itself not properly working resulting in this error which does not seem connected from the first sight. The difference is that I use "build", not "export, not sure if it can help you then :(
